src/basic_recipes/voronoiplot.jl

```diff
@@ -0,0 +1,171 @@
+- `cycle = [:color => :patchcolor]` sets the cycle for the polygons.
+"""
+@recipe(Voronoiplot, vorn) do scene
+    th = default_theme(scene, Mesh)
+    sc = default_theme(scene, Scatter)
+    return Attributes(;
+                      # Toggles
+                      show_generators=true,
+                      smooth=false,
+
+                      # Point settings
+                      markersize=4,
+                      marker=sc.marker,
+                      point_color=sc.color,
+
+                      # Polygon settings
+                      strokecolor=theme(scene, :patchstrokecolor),
+                      strokewidth=1.0,
+                      polygon_color=automatic,
+                      unbounded_edge_extension_factor=0.1,
+                      bounding_box=automatic,
+
+                      # Colormap settings
+                      colormap=th.colormap,
+                      colorrange=th.colorrange,
+                      cycle=th.cycle)
+end
+
+function get_voronoi_tiles!(generators, polygons, vorn, bbox)
+    empty!(generators)
+    empty!(polygons)
+    sizehint!(generators, DelTri.num_generators(vorn))
+    sizehint!(polygons, DelTri.num_polygons(vorn))
+    if !isnothing(bbox)
+        a, b, c, d = bbox
+        @assert a < b && c < d "Bounding box must be of the form (xmin, xmax, ymin, ymax)."
+    end
+    for i in DelTri.each_generator(vorn)
+        g = DelTri.get_generator(vorn, i)
+        x, y = DelTri.getxy(g)
+        push!(generators, Point2f(x, y))
+        !isnothing(bbox) &&
+            @assert a ≤ x ≤ b && c ≤ y ≤ d "Generator $(i) with coordinates ($x, $y) is outside the bounding box."
+        polygon_coords = DelTri.get_polygon_coordinates(vorn, i, bbox)
+        polygon_coords_2f = map(polygon_coords) do coords
+            x, y = DelTri.getxy(coords)
+            !isnothing(bbox) &&
+                @assert a ≤ x ≤ b && c ≤ y ≤ d "Polygon vertex $(i) with coordinates ($x, $y) is outside the bounding box."
+            return Point2f(x, y)
+        end
+        push!(polygons, Polygon(polygon_coords_2f))
+    end
+    return generators, polygons
+end
+
+function get_voronoi_colors!(colors, vorn, cmap)
+    empty!(colors)
+    sizehint!(colors, DelTri.num_polygons(vorn))
+    F = DelTri.number_type(vorn)
+    gtr = [DelTri.get_generator(vorn, i) for i in DelTri.each_generator(vorn)]
+    reverse!(gtr) # For some reason this is needed to get distinct colors for the tiles
+    gtr_mat = reinterpret(reshape, F, gtr)
+    _colors = get(cgrad(cmap), gtr_mat, :extrema)
+    for c in eachcol(_colors)
+        a, b = c
+        push!(colors, (a + b) / 2)
+    end
+    return colors
+end
+
+
+Makie.conversion_trait(::Voronoiplot) = PointBased()
+Makie.convert_arguments(::Voronoiplot, x::DelTri.VoronoiTessellation) = (x,)
+
+function Makie.plot!(p::Voronoiplot{<: Tuple{<: Vector{<: Point2f}}})
```

Comment:
These three function are there so `voronoiplot` also works with PointBased data, so `Vector, Vector` or `Vector{Point2f}`. I guess we would want this for triplot too?

Reply:
Yes, I'll put those into `triplot` on Monday. Basically anything that works for `voronoiplot` should work for `triplot`.

Reply:
I remembered how this works incorrectly but it should work now